### PR TITLE
Add python project gitignore

### PR DESCRIPTION
**Pull request by yurinnick**
Use default Github `.gitignore` template for python projects

### CODE REVIEW

**Review by gctucker — COMMENTED** [review]
Why would you do this?

**Comment by yurinnick** [comment]
After a few day working with `kernelci-core` there are temporary python files created, that are not ignored:
```
$ git check-ignore **/*
build/bdist.linux-x86_64
build/lib
build/scripts-3.11
kernelci.egg-info/dependency_links.txt
kernelci.egg-info/PKG-INFO
kernelci.egg-info/requires.txt
kernelci.egg-info/SOURCES.txt
kernelci.egg-info/top_level.txt
kernelci/__pycache__
```

**Comment by gctucker** [comment]
Ah yes, I guess that falls into the category of making this a proper Python package.

**Review by nuclearcat — APPROVED** [review]
Not sure if *.so is good here, but i think we should avoid using .so in project anyway, verified "check-ignore" at my side, LGTM.

**Comment by gctucker** [comment]
> Not sure if *.so is good here, but i think we should avoid using .so in project anyway, verified "check-ignore" at my side, LGTM.

It's here in case we have any native Python modules implemented in C I guess.  And yes I don't see a use-case where we would keep executable binaries under source control in an LGPL project anyway.

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,160 @@
...
+lib64/
+parts/
+sdist/
+var/
+wheels/
+share/python-wheels/
+*.egg-info/
+.installed.cfg
+*.egg
+MANIFEST
```
Comment (gctucker) [inline .gitignore:27]: Is this always meant to be an automatically-generated file?

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,160 @@
...
+.hypothesis/
+.pytest_cache/
+cover/
+
+# Translations
+*.mo
+*.pot
+
+# Django stuff:
+*.log
```
Comment (gctucker) [inline .gitignore:59]: This one might bite us at some point in the future, for example if we wan to keep sample log files in the documentation.  Probably we can just remove the Django section.

File: .gitignore [hunk 3]
```diff
@@ -1,2 +1,160 @@
...
+celerybeat-schedule
+celerybeat.pid
+
+# SageMath parsed files
+*.sage.py
+
+# Environments
+.env
+.venv
+env/
```
Comment (gctucker) [inline .gitignore:125]: In other projects there are `env` directories to keep sample environment files, or encrypted actual ones used in deployments.  While ignoring `.env` seems safe, I'm not sure `env/` is a good idea.

**Review by gctucker — COMMENTED** [review]
A few things to consider.  The main risk here is to not have files added to PRs by mistake because of zealous gitignore settings.  Ideally, I think we should only be adding only entries for the files we positively don't want to have in git rather than blindly applying some default list.

File: .gitignore (hunk 1 above)
Comment (yurinnick) [reply .gitignore:27]: Based on [python documentation](https://docs.python.org/3/distutils/sourcedist.html) - yes:

```
The manifest template is just a list of instructions for how to generate your manifest file, MANIFEST, which is the exact list of files to include in your source distribution. The sdist command processes this template and generates a manifest based on its instructions and what it finds in the filesystem.
```

**Comment by yurinnick** [comment]
Remove Django section and `env` directory from `.gitignore`

File: .gitignore (hunk 1 above)
Comment (gctucker) [reply .gitignore:27]: Ah yes thanks, I think I confused it with the manifest template.

**Review by gctucker — APPROVED** [review]

**Comment by yurinnick** [comment]
@gctucker I don't have permissions to merge, so if it looks good, please merge it.

**Pull request merged**